### ISSUE TITLE
Transformers in Vision: A Survey

### ISSUE THREAD

**Issue by yskim0** [yskim0/One-Day-One-Paper#157]
**page 10 of 28까지 읽고 정리함**

🔍 &nbsp;CNN과 다르게 Transforme의 두드러지는 점은 minimal inductive bias를 사용한다는 점
🔍 &nbsp;트랜스포머는 **minimal** prior knowledge을 가지고 있기 때문에 주로 large-scale dataset에 대해 **pretext task**를 활용하여 사전학습된다. (self-supervised learning)

🔍   &nbsp;Foundations
- self-attn
- pre-training

🔍   &nbsp;pre-training --> self-supervised learning
- `fill in the blank`
- `contrastive learning`
  - e.g., replacing an object with another in the same scene, or changing the class with minor adversarial changes to the image
- SSL methods --> (a) generative approaches which synthesize~ (b) context-based methods which exploit the relationships between image patches or video frames (c) cross-modal methods which leverage from multiple data modalities
  - (b) temporal sequence of video !! 

🔍  &nbsp;Bidirectional Representations
- BERT
  - Masked Language Model
  - Next Sentence Prediction

[A taxonomy of self-attention design space]

![IMG_47D21CE3F556-1](https://user-images.githubusercontent.com/48315997/133991030-9d78741e-c5cd-4618-9413-3c9622a81a33.jpeg)

🔍  &nbsp; Attention Augmented Convolutional Networks
> self-attention as an alternative to convolutional operators. They propose to use the relative position encoding in two dimensions to develop a new self-attn mechanism

🔍  &nbsp; Vectorized Self-Attention
- pairwise self-attention
- patch-wise self-attention

🔍  &nbsp; CLIP : Contrastive Language-Image Pre-training
> CLIP is a contrastive approach to learn image representations from text, with a learning objective which maximizes similarity of correct text-image pairs embedding in a large batch size

**🔍  &nbsp; Transformers for Object Detection**
- DETR
  - treats object detection as a set prediction problem and proposes a set loss function
  - 1) prediction a set of objects
  - 2) allows bipartite matching between prediction & g.t.
  - RPN, NMS 필요 없어졌다

**🔍  &nbsp; Transformers for Image Generation**
- Image Transformer
  - sequentially predict each pixel of an output image given its previously generated pixels
  - Memory block, query block
